File: src/main/java/com/example/demo/model/BlacklistEvent.java
package com.example.demo.model;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.OffsetDateTime;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class BlacklistEvent {
    private Long id;
    private OffsetDateTime createdAt;
    private String requestId;
    private String ipAddress;
    private String method;
    private String path;
    private Integer status;
    private String userAgent;
    private String referer;
    private String reason;
    private String source;
}

File: src/main/kotlin/com/example/demo/model/BlacklistEvent.kt
package com.example.demo.model

import java.time.OffsetDateTime

data class BlacklistEvent(
    val id: Long? = null,
    val createdAt: OffsetDateTime? = null,
    val requestId: String? = null,
    val ipAddress: String,
    val method: String? = null,
    val path: String? = null,
    val status: Int? = null,
    val userAgent: String? = null,
    val referer: String? = null,
    val reason: String,
    val source: String,
)

